apps/prairielearn/src/components/InstructorInfoPanel.html.ts

```diff
@@ -0,0 +1,372 @@
+}) {
+  // Here, we are only checking if the effective user is an instructor. We are not
+  // attempting to check if this user has permission to view student data. That check
+  // would have already been made (when necessary) before granting access to the page
+  // on which this partial is included (an assessment instance or question instance).
+  if (
+    !authz_data.has_course_permission_preview &&
+    !authz_data.has_course_instance_permission_view
+  ) {
+    return '';
+  }
+
+  return html`
+    <div class="card mb-4 border-warning">
+      <div class="card-header bg-warning">Staff information</div>
+      <div class="card-body">
+        ${StaffUserInfo({ user })}
+        ${InstanceUserInfo({ instance_user, instance_group, instance_group_uid_list })}
+        ${QuestionInfo({ course, course_instance, question, variant, question_is_shared })}
+        ${VariantInfo({ variant })} ${IssueReportButton({ variant, csrfToken, questionContext })}
+        ${AssessmentInstanceInfo({ assessment, assessment_instance })}
+        ${ManualGradingInfo({ instance_question, assessment, questionContext })}
+      </div>
+      <div class="card-footer small">This box is not visible to students.</div>
+    </div>
+  `;
+}
+
+function StaffUserInfo({ user }: { user: User }) {
+  return html`
+    <h5 class="card-title">Staff user:</h5>
+    <div class="d-flex flex-wrap pb-2">
+      <div class="pr-1">${user.name}</div>
+      <div class="pr-1">${user.uid}</div>
+    </div>
+  `;
+}
+
+function InstanceUserInfo({
+  instance_user,
+  instance_group,
+  instance_group_uid_list,
+}: {
+  instance_user?: User | null;
+  instance_group?: Group | null;
+  instance_group_uid_list?: string[] | null;
+}) {
+  if (instance_user == null && instance_group == null) return '';
+  return html`
+    <hr />
+    <div>
+      <details>
+        ${instance_group != null
+          ? html`
+              <summary><h5 class="card-title">Group details</h5></summary>
+              <div class="d-flex flex-wrap pb-2">
+                <div class="pr-1">${instance_group.name}</div>
+                <div class="pr-1">(${instance_group_uid_list?.join(', ')})</div>
+              </div>
+            `
+          : html`
+              <summary>
+                <h5 class="card-title d-inline-block">Student details</h5>
+              </summary>
+              <div class="d-flex flex-wrap pb-2">
+                <div class="pr-1">${instance_user?.name}</div>
+                <div class="pr-1">${instance_user?.uid}</div>
+              </div>
+            `}
+      </details>
+    </div>
+  `;
+}
+
+function QuestionInfo({
+  course,
+  course_instance,
+  question,
+  variant,
+  question_is_shared,
+}: {
+  course: Course;
+  course_instance?: CourseInstance;
+  question?: Question;
+  variant?: Variant;
+  question_is_shared?: boolean;
+}) {
+  if (question == null) return '';
+
+  const variantSeedQuery = variant != null ? `?variant_seed=${variant.variant_seed}` : '';
```

Comment:
Good catch. Since this variable is also only used once I rearranged it to make it clearer. This check seems to have been introduced in #9926 as an extra precaution that isn't actually necessary, so it seems the rewriting of the code made this cleaner.